File: Service/Models/API/Match.cs
using System;

namespace BetSnooker.Models.API
{
    public class Match
    {
        public int Id { get; set; } // TODO: this ID changes with requests
        public int WorldSnookerId { get; set; } // TODO: this is probably useless (verify on live event)
        public int EventId { get; set; }
        public int Round { get; set; }
        public int Number { get; set; }
        public int Player1Id { get; set; }
        public int Score1 { get; set; }
        public bool Walkover1 { get; set; }
        public int Player2Id { get; set; }
        public int Score2 { get; set; }
        public bool Walkover2 { get; set; }
        public int WinnerId { get; set; }
        public bool Unfinished { get; set; }
        public bool OnBreak { get; set; }
        public DateTime? ScheduledDate { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string Sessions { get; set; }
    }
}

File: Web.UI/src/app/_models/match.ts
export class Match {
    matchId: number;
    eventId: number;
    round: number;
    roundName: string;
    distance: number;
    number: number;
    player1Id: number;
    player1Name: string;
    score1: number;
    walkover1: boolean;
    player2Id: number;
    player2Name: string;
    score2: number;
    walkover2: boolean;
    winnerId: number;
    winnerName: string;
    unfinished: boolean;
    onBreak: boolean;
    scheduledDate: Date;
    startDate: Date;
    endDate: Date;
    actualStartDate: Date;
}
